### DISCUSSION TITLE
CI Statuses: current UI doesn't work with 16 statuses

### DISCUSSION THREAD

**Discussion by danielbeardsley** [iFixit/pulldasher#316] [Q&A] [answered]
The current UI works when we have a small number of CI statuses (maybe < 8), but is really annoying with anything more than that.

We need to find a better way to expose the CI statuses.

![image](https://user-images.githubusercontent.com/26855/168345035-dc5750e0-39e2-4b16-8339-bb819558604e.png)

* Problem: Targets are far too small and therefore the contextual info we need (which ones are failing, link to the log) is really hard to access.
  * Note: I only ever click on the not-green CI statuses
* Note: CI status links, branch name copying, and click-to-open-pulls are the only clickable things

**Comment by danielbeardsley** [comment]
Potential Solution

Swap the position of the Age and CI status indicators because there's more horizontal room

**Reply by mlahargou** [reply]
This seems like the easiest option. Although we will run into the same issue if we have 100 statuses.

**Comment by danielbeardsley** [comment]
Potential Solution

Group all the successful CI statuses into *one* green bubble that has a popup with more info

**Reply by davidrans** [reply]
I think this is the same thing as I suggested below. I'll move my mockup here:
![image](https://user-images.githubusercontent.com/806152/168347191-0b51f7e7-8b78-4f04-a55b-a1ac45a39d90.png)

**Reply by danielbeardsley** [reply]
Are you saying we show all the non-green bubbles but just have one that's green to represent all of green ones?

**Reply by davidrans** [reply]
Yeah, I thought that'd be an easy way to stick with something close the current UI

**Comment by danielbeardsley** [comment, marked as answer]
Potential Solution

Group statuses by state so there are never more than 4, each with their own popup which shows more info (context and link to logs)

**Reply by mlahargou** [reply]
I think this is the most generic and most likely to work for any number of statuses.

**Reply by danielbeardsley** [reply]
I've got a rough working version, feel it out at https://pulldasher-dev.cominor.com/

![image](https://user-images.githubusercontent.com/26855/168395988-337dc1f3-2125-4137-b33c-64ada30f3809.png)

**Reply by davidrans** [reply]
Nice! Looks good to me

**Reply by danielbeardsley** [reply]
I polished the styling and will open a pull:
![image](https://user-images.githubusercontent.com/26855/168635060-502f1b14-4664-4cd5-8608-e4caf4bc401c.png)

**Reply by danielbeardsley** [reply]
Pull open #318

**Comment by mlahargou** [comment]
If we do any sort of grouping, we might want to add a visible count as well. I treat PRs that have 6 failures differently than PRs that have 1 failure.

**Reply by danielbeardsley** [reply]
We could roughly scale the visual size (with some minimum) based on how many there are:
![image](https://user-images.githubusercontent.com/26855/168347272-214aefc2-f7f2-444b-bbc7-eb4ec7dd96bb.png)

i.e. that's 6 greens, a red, 1 gray, and 2 oranges. The groups are slightly scaled based on how many they represent.

**Reply by mlahargou** [reply]
Yeah that works!

**Reply by davidrans** [reply]
I like this idea too, I just voted for https://github.com/iFixit/pulldasher/discussions/316#discussioncomment-2747788 cause I think it's less work

**Comment by mlahargou** [comment]
In any solution, can we sort the statuses by context? It's frustrating when they show up in a random order.

**Comment by davidrans** [comment]


**Reply by mlahargou** [reply]
That's Danny's second suggestion I think.

**Reply by davidrans** [reply]
Yeah, closing this one in favor of that one

**Comment by jarstelfox** [comment]
Potential Solution

Use https://chakra-ui.com/docs/components/disclosure/accordion and *just* display the overall status, expand to see all statuses

**Reply by danielbeardsley** [reply]
Expanding on click sounds like a good idea. Can you be a little more concrete with perhaps a rough drawing?

**Reply by jarstelfox** [reply]
Sure. I can't get any online tool working nicely let me grab a pen and paper.

**Reply by jarstelfox** [reply]
Expand on click:

![8F18AEA3-6D00-495C-94F9-EE4752CB3EE7](https://user-images.githubusercontent.com/857362/168373737-d6291fae-06a8-4026-8620-3dde5392ba66.jpeg)

The log links could be labeled since there is enough space

**Reply by jordycosta** [reply]
This is cool!

**Comment by andyg0808** [comment]
I think this might be solved in https://github.com/iFixit/pulldasher/pull/318 and https://github.com/iFixit/pulldasher/pull/322

**Reply by danielbeardsley** [reply]
Huh... I guess discussions don't have a "closed" state. 

https://github.com/github/feedback/discussions/3097
